# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [12]:
s1 = pd.Series(np.random.randint(1, 4, 100))
s2 = pd.Series(np.random.randint(1, 3, 100))
s3 = pd.Series(np.random.randint(10000, 30000, 100))

print(s1)
print(s2)
print(s3)

0     1
1     1
2     2
3     3
4     2
     ..
95    1
96    3
97    1
98    3
99    3
Length: 100, dtype: int64
0     1
1     2
2     2
3     1
4     1
     ..
95    1
96    1
97    1
98    2
99    1
Length: 100, dtype: int64
0     19327
1     11636
2     19822
3     15150
4     29591
      ...  
95    12504
96    20607
97    11723
98    23746
99    21153
Length: 100, dtype: int64


### Step 3. Let's create a DataFrame by joinning the Series by column

In [29]:
df = pd.concat([s1, s2, s3], axis=1, ignore_index=True)
print(df)

    0  1      2
0   1  1  19327
1   1  2  11636
2   2  2  19822
3   3  1  15150
4   2  1  29591
.. .. ..    ...
95  1  1  12504
96  3  1  20607
97  1  1  11723
98  3  2  23746
99  3  1  21153

[100 rows x 3 columns]


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [30]:
df.columns = ["bedrs", "bathrs", "price_sqr_meter"]
print(df)

    bedrs  bathrs  price_sqr_meter
0       1       1            19327
1       1       2            11636
2       2       2            19822
3       3       1            15150
4       2       1            29591
..    ...     ...              ...
95      1       1            12504
96      3       1            20607
97      1       1            11723
98      3       2            23746
99      3       1            21153

[100 rows x 3 columns]


### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [ ]:
big_column = df.stack().values
new_df = pd.DataFrame({"bigcolumn" : big_column})
print(new_df)

     bigcolumn
0            1
1            1
2        19327
3            1
4            2
..         ...
295          2
296      23746
297          3
298          1
299      21153

[300 rows x 1 columns]


### Step 6. Oops, it seems it is going only until index 99. Is it true?

### Step 7. Reindex the DataFrame so it goes from 0 to 299